qa/workunits/cephadm/test_adoption.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash -ex
+
+SCRIPT_NAME=$(basename ${BASH_SOURCE[0]})
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+CEPHADM_SRC_DIR=${SCRIPT_DIR}/../../../src/cephadm
+
+CORPUS_GIT_SUBMOD="cephadm-adoption-corpus"
+TMPDIR=$(mktemp -d)
+git clone https://github.com/ceph/$CORPUS_GIT_SUBMOD $TMPDIR
+CORPUS_DIR=${TMPDIR}/archive
+trap "$SUDO rm -rf $TMPDIR" EXIT
+
+[ -z "$SUDO" ] && SUDO=sudo
+
+if [ -z "$CEPHADM" ]; then
+    CEPHADM=${CEPHADM_SRC_DIR}/cephadm
+fi
+
+# at this point, we need $CEPHADM set
+if ! [ -x "$CEPHADM" ]; then
+    echo "cephadm not found. Please set \$CEPHADM"
+    exit 1
+fi
+
+# respawn ourselves with a shebang
+if [ -z "$PYTHON_KLUDGE" ]; then
+    # see which pythons we should test with
+    PYTHONS=""
+    which python3 && PYTHONS="$PYTHONS python3"
+    which python2 && PYTHONS="$PYTHONS python2"
+    echo "PYTHONS $PYTHONS"
+    if [ -z "$PYTHONS" ]; then
+	echo "No PYTHONS found!"
+	exit 1
+    fi
+
+    TMPBINDIR=$(mktemp -d)
+    trap "rm -rf $TMPBINDIR" EXIT
+    ORIG_CEPHADM="$CEPHADM"
+    CEPHADM="$TMPBINDIR/cephadm"
+    for p in $PYTHONS; do
+	echo "=== re-running with $p ==="
+	ln -s `which $p` $TMPBINDIR/python
+	echo "#!$TMPBINDIR/python" > $CEPHADM
+	cat $ORIG_CEPHADM >> $CEPHADM
+	chmod 700 $CEPHADM
+	$TMPBINDIR/python --version
+	PYTHON_KLUDGE=1 CEPHADM=$CEPHADM $0
+	rm $TMPBINDIR/python
+    done
+    rm -rf $TMPBINDIR
+    echo "PASS with all of: $PYTHONS"
+    exit 0
+fi
+
+## adopt
+for subdir in `ls ${CORPUS_DIR}`; do
+    for tarfile in `ls ${CORPUS_DIR}/${subdir} | grep .tgz`; do
+	tarball=${CORPUS_DIR}/${subdir}/${tarfile}
+	FSID_LEGACY=`echo "$tarfile" | cut -c 1-36`
```

Comment:
This works, but it's a little cumbersome. Maybe get the fsid from `cephadm ls` below?